### PR TITLE
Use punycode more appropriately

### LINKED ISSUE #5: Wrong answer on "\u001fx", apparently
Continued from https://github.com/w3c/web-platform-tests/pull/4504#issuecomment-270771165 . Repro:

```js
"use strict";
const tr46 = require(".");
const util = require("util");

const input = "\u001fx";
const output = tr46.toASCII(input, false, tr46.PROCESSING_OPTIONS.TRANSITIONAL, false);
console.log(util.inspect(output));
```

outputs `'xn--\u001fx-'` whereas it should output just `'\u001fx'` according to @annevk's reading in https://github.com/w3c/web-platform-tests/pull/4504#issuecomment-270848858 . Updates to follow...

### PR DESCRIPTION
**Pull request by domenic**
Previously we used punycode's toASCII and toUnicode exports, which implement incomplete parts of TR46 themselves (with buggy "is ASCII" checks until recently; see #5 and https://github.com/bestiejs/punycode.js/issues/59. Now we usethe lower-level encode and decode exports, with appropriate tweaks to the surrounding code to more fully conform to the surrounding TR46 algorithm steps.

Fixes #5.

---

Still 488 failing tests before and after. However this improves jsdom/whatwg-url results on the host parser tests at https://github.com/w3c/web-platform-tests/pull/4504 to only 3 failures none of which seem tr46 related.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -12,6 +12,16 @@ function normalize(str) { // fix bug in v8
   return str.split('\u0000').map(function (s) { return s.normalize('NFC'); }).join('\u0000');
 }
 
+function containsNonASCII(str) {
```
Comment (Sebmaster) [inline index.js]: Regexes seem to be [much faster for this](https://jsperf.com/ascii-regex-vs-loop).

Comment (domenic) [inline index.js]: Dang, I should have checked.

**Comment by Sebmaster** [comment]
Regarding the test failures: A lot of that is just the inadequate description of the test suite by Unicode. Getting failures down is basically only a heuristic.

**Comment by Sebmaster** [comment]
Should also switch to punycode.js proper (the package) instead of using the node built-in module while we're at it.

**Pull request merged**